### PR TITLE
File Hw_ 2_ add.py is a file with an additional howework #2.

### PR DESCRIPTION
**Pull request by Oleksiy-Yurchenko**
File contains 3 functions.

### CODE REVIEW

File: hw_2_add.py [hunk 1]
```diff
@@ -4,7 +4,9 @@ def smallest_integer(array):
     else:
         arr_max = max(array)
         return min(set(array).intersection(set(range(1,arr_max + 1)))) if arr_max > 0 else 1
-    
+
```
Comment (mmykhailenko) [inline hw_2_add.py:5]: this function works incorrectly. Pls check with examples in condition 
For example, given A = [1, 3, 6, 4, 1, 2], the function should return 5.
Given A = [1, 2, 3], the function should return 4.
Given A = [−1, −3], the function should return 1.

Comment (mmykhailenko) [reply hw_2_add.py:5]: binary_parser - also works incorrectly

**Review by mmykhailenko — COMMENTED** [review]
General comment. Give more clear names for variables. Check the code before you push it. Remove any other local files from PR like .idea/, Scripts..

File: hw_2_add.py (hunk 1 above)
Comment (Oleksiy-Yurchenko) [reply hw_2_add.py:5]: Thanks for review. All resolved.

**Pull request merged**